src/ui/hooks.ts

```diff
@@ -221,3 +221,28 @@ export function useControlledState<T>(
 
     return [state, setStateWithUpdate];
 }
+/**
+ * a simple hook that leverages `useEffect` and `useState` to
+ * return some async data and its fulfillment status.
+ *
+ * @group Hooks
+ * @param loader a parameterless function that returns a promise
+ * @param deps optional deps to pass to useEffect
+ * @returns a tuple containing the resolved promise's value
+ * and a boolean indicating whether the promise has resolved.
+ */
+export function useAsync<T>(loader: () => Promise<T>, deps: any[] = []): [T, boolean] {
```

Comment:
I'm not sure this is race condition safe as is. Say I have something like this:

-> first useAsync call starts (promise A)
-> second useAsync call starts (promise B)
-> promise B resolves, updates value
-> promise A resolves, updates values

Then the original load will clobber the later load. You should add a generation number or similar state in a `useRef` which prevents earlier slow loads from overwriting later loads.